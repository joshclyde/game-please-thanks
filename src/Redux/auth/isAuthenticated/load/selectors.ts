import { makeUseSelector } from "@ReduxUtils";

import { RootState } from "../../../types";

export const useSelectIsAuthenticationLoading = makeUseSelector(
  () => (state: RootState) => state.auth.isAuthenticated.load.isPending,
);
export const useSelectDidAuthenticationLoadSucceed = makeUseSelector(
  () => (state: RootState) => state.auth.isAuthenticated.load.isSuccess,
);
export const useSelectDidAuthenticationLoadFail = makeUseSelector(
  () => (state: RootState) => state.auth.isAuthenticated.load.isFailure,
);
