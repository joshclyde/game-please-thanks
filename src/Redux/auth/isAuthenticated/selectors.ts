import { RootState } from "@Redux";
import { makeUseSelector } from "@ReduxUtils";

const makeSelectIsAuthenticated = () => (state: RootState) => state.auth.isAuthenticated;

export const useSelectIsAuthenticated = makeUseSelector(makeSelectIsAuthenticated);
