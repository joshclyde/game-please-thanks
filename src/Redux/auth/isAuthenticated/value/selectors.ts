import { RootState } from "@Redux";
import { makeUseSelector } from "@ReduxUtils";

export const makeSelectIsAuthenticated = () => (state: RootState) =>
  state.auth.isAuthenticated.value;

export const useSelectIsAuthenticated = makeUseSelector(makeSelectIsAuthenticated);
