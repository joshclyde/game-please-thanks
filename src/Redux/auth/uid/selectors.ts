import { RootState } from "@Redux";
import { makeUseSelector } from "@ReduxUtils";

export const makeSelectUid = () => (state: RootState) => state.auth.uid;

export const useSelectUid = makeUseSelector(makeSelectUid);
