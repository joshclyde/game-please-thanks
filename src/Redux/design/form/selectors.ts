import { makeUseSelector } from "@ReduxUtils";

import { RootState } from "../../types";

const makeSelectDoesFormExist = (formId: string) => (state: RootState) =>
  Boolean(state.design.form[formId]);

export const useSelectDoesFormExist = makeUseSelector(makeSelectDoesFormExist);

const makeSelectFormValues = (formId: string) => (state: RootState) =>
  state.design.form[formId];

export const useSelectFormValues = makeUseSelector(makeSelectFormValues);

const makeSelectFormInputValue = (formId: string, inputId: string) => (
  state: RootState,
) => state.design.form[formId]?.[inputId];

export const useSelectFormInputValue = makeUseSelector(makeSelectFormInputValue);
