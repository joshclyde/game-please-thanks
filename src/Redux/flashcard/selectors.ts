import { State } from "../types";

// TODO: look into ReadOnly typescript for selectors?

export type StateJustFlashcard = Pick<State, "flashcard">;

export const selectFlashcardData = (state: StateJustFlashcard) => state.flashcard.data;

export const selectFlashcardDataSetIds = (state: StateJustFlashcard) =>
  Object.keys(state.flashcard.data);

export const selectFlashcardSetTitle = (state: StateJustFlashcard, setId: string) =>
  state.flashcard.data[setId].title;

export const selectFlashcardQuizCurrentId = (state: StateJustFlashcard) =>
  state.flashcard.quiz.id;

export const selectFlashcardQuizLength = (state: StateJustFlashcard) =>
  state.flashcard.quiz.cardResponses &&
  Object.keys(state.flashcard.quiz.cardResponses).length;

export const selectFlashcardQuizCurrentCardId = (state: StateJustFlashcard) =>
  state.flashcard.quiz.currentCard;

export const selectFlashcardQuizFlashcards = (state: StateJustFlashcard) =>
  state.flashcard.quiz.cardResponses &&
  Object.keys(state.flashcard.quiz.cardResponses).reduce<
    Record<
      string,
      {
        question: string;
        answer: string;
        isCorrect: boolean;
      }
    >
  >(
    (flashcardContents, flashcardKey) => ({
      ...flashcardContents,
      [flashcardKey]: {
        ...state.flashcard.quiz.cardResponses[flashcardKey],
        ...state.flashcard.data[state.flashcard.quiz.id].flashcards[flashcardKey],
      },
    }),
    {},
  );
//   (flashcardContents, flashcardKey) => {
//       ...flashcardContents,
//       [flashcardKey]: {}
//   },
// {}: any; )
