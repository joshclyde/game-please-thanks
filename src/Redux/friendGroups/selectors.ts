import { makeUseSelector } from "@ReduxUtils";

import { useSelectCurrentFriendGroupId } from "../settings/hooks";
import { RootState } from "../types";

const makeSelectFriendGroupGameIds = (friendGroupId: string) => (state: RootState) =>
  state.friendGroups[friendGroupId]?.games;

export const useSelectFriendGroupGameIds = makeUseSelector(makeSelectFriendGroupGameIds);

export const useSelectCurrentFriendGroupGameIds = () => {
  const groupId = useSelectCurrentFriendGroupId();
  const gameIds = useSelectFriendGroupGameIds(groupId);
  return gameIds;
};
