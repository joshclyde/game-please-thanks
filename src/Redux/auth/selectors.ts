import _ from "lodash";

import { RootState } from "@Redux";
import { makeUseSelector } from "@ReduxUtils";

import {
  makeSelectUserName,
  makeSelectUserHasGamePass,
  makeSelectUserGamesOwnedIds,
  makeSelectUserFriends,
  makeSelectUserFriendsIds,
  makeSelectUserIsGameOwned,
} from "../users/selectors";

import { makeSelectUid } from "./uid/selectors";

/*
  name
*/
const makeSelectAuthName = () => (state: RootState) => {
  const uid = makeSelectUid()(state);
  return uid ? makeSelectUserName(uid)(state) : null;
};

export const useSelectAuthName = makeUseSelector(makeSelectAuthName);

/*
  hasGamePass
*/
const makeSelectAuthHasGamePass = () => (state: RootState) => {
  const uid = makeSelectUid()(state);
  return uid ? makeSelectUserHasGamePass(uid)(state) : null;
};

export const useSelectAuthHasGamePass = makeUseSelector(makeSelectAuthHasGamePass);

/*
  games
*/
const makeSelectAuthGamesOwnedIds = () => (state: RootState) => {
  const uid = makeSelectUid()(state);
  return uid ? makeSelectUserGamesOwnedIds(uid)(state) : null;
};

export const useSelectProfileGamesOwnedIds = makeUseSelector(makeSelectAuthGamesOwnedIds);

const makeSelectAuthIsGameOwned = (gameId: string) => (state: RootState) => {
  const uid = makeSelectUid()(state);
  return uid ? makeSelectUserIsGameOwned(uid, gameId)(state) : null;
};

export const useSelectAuthIsGameOwned = makeUseSelector(makeSelectAuthIsGameOwned);

/*
  friends
*/
const makeSelectAuthFriends = () => (state: RootState) => {
  const uid = makeSelectUid()(state);
  return uid ? makeSelectUserFriends(uid)(state) : null;
};

export const useSelectAuthFriends = makeUseSelector(makeSelectAuthFriends);

const makeSelectAuthFriendsIds = () => (state: RootState) => {
  const uid = makeSelectUid()(state);
  return uid ? makeSelectUserFriendsIds(uid)(state) : null;
};

export const useSelectAuthFriendsIds = makeUseSelector(makeSelectAuthFriendsIds);

const makeSelectAuthFriendsIdsSorted = () => (state: RootState) => {
  const friends = makeSelectAuthFriends()(state);
  return friends
    ? Object.keys(friends).sort((friendId1, friendId2) => {
        return friends[friendId1].name.localeCompare(friends[friendId2].name);
      })
    : null;
};

export const useSelectAuthFriendIdsSorted = makeUseSelector(
  makeSelectAuthFriendsIdsSorted,
);

const makeSelectAuthFriendIdsThatOwnGame = (gameId: string) => (state: RootState) => {
  const friends = makeSelectAuthFriends()(state);
  const friendsThatOwnGame = _.pickBy(friends, (user) =>
    Boolean(user.games?.[gameId]?.isOwned),
  );
  return Object.keys(friendsThatOwnGame);
};

export const useSelectAuthFriendIdsThatOwnGame = makeUseSelector(
  makeSelectAuthFriendIdsThatOwnGame,
);
