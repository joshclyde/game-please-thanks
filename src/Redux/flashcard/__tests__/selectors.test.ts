import { selectFlashcardQuizFlashcards } from "../selectors";

import { mockFlashcardA } from "./mockData";

describe(`selectors`, () => {
  // test.each(Object.keys(selectors))("selector: %s", (selector) => {
  //   // @ts-ignore
  //   expect(selectors[selector](mockData)).toMatchSnapshot();
  // });

  describe(`selectFlashcardQuizFlashcards`, () => {
    test(`should return merged flashcard data and quiz data`, () => {
      expect(selectFlashcardQuizFlashcards(mockFlashcardA)).toMatchSnapshot();
    });
  });

  // test("selectFlashcardQuizFlashcards", () => {
  //   expect(selectFlashcardQuizFlashcards(mockData)).toMatchSnapshot();
  // });
});
