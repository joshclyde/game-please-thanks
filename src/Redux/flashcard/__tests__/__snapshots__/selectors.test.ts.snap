// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors selectFlashcardQuizFlashcards should return merged flashcard data and quiz data 1`] = `
Object {
  "a1": Object {
    "answer": "a1 answer",
    "isCorrect": true,
    "question": "a1 question",
  },
  "a2": Object {
    "answer": "a2 answer",
    "isCorrect": true,
    "question": "a2 question",
  },
}
`;
